src/Sylius/Behat/Context/Api/Admin/ManagingTaxRateContext.php

```diff
@@ -0,0 +1,524 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api\Admin;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Context\Api\Resources;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Core\Model\TaxRateInterface;
+use Sylius\Component\Taxation\Model\TaxCategoryInterface;
+use Webmozart\Assert\Assert;
+
+class ManagingTaxRateContext implements Context
+{
+    public function __construct(
+        private ApiClientInterface $client,
+        private ResponseCheckerInterface $responseChecker,
+        private IriConverterInterface $iriConverter,
+        private SharedStorageInterface $sharedStorage
+    ) {
+    }
+
+    /**
+     * @When I want to create a new tax rate
+     */
+    public function iWantToCreateANewTaxRate(): void
+    {
+        $this->client->buildCreateRequest(Resources::TAX_RATES);
+    }
+
+    /**
+     * @When I specify its code as :code
+     * @When I do not specify its code
+     */
+    public function iSpecifyItsCodeAs(?string $code = null): void
+    {
+        if ($code) {
+            $this->client->addRequestData('code', $code);
+        }
```

Comment:
```suggestion
        if (null === $code) {
            return;
        }
        
        $this->client->addRequestData('code', $code);
```
I'd be for using an early return and `null === $code` in place of just a `$code`.
If you accept my suggestion, the same thing should be done with other similar methods.

Reply:
but going forward with this logic, we could put all these empty steps with one placeholder action and not perform the check at all. It is a waste of electricity. At the moment of step match without argument, with the current implementation, we are sure nothing will happen.

The question is if we shouldn't think about the `When I do not specify its code` step as a step, where we are filing an empty string into the body of the request.

Reply:
@lchrusciel for string I removed the possibility to set it as null. I replaced the default value with an empty string and just set it (I've found it in one of other API Behat Contexts, and it makes sense). For Zone I left the null value checking. It'll be pushed along the other fixes.